frontend/app/src/components/settings/general/TimeFrameSettings.vue

```diff
@@ -95,138 +95,159 @@ import {
   TimeFramePersist,
   TimeFrameSetting
 } from '@rotki/common/lib/settings/graphs';
-import { Component, Emit, Mixins, Prop } from 'vue-property-decorator';
-import { mapMutations } from 'vuex';
+import {
+  computed,
+  defineComponent,
+  PropType,
+  toRefs
+} from '@vue/composition-api';
+import { get } from '@vueuse/core';
 import Fragment from '@/components/helper/Fragment';
-import TimeframeSelector from '@/components/helper/TimeframeSelector.vue';
-import PremiumMixin from '@/mixins/premium-mixin';
-import SettingsMixin from '@/mixins/settings-mixin';
+import { getPremium } from '@/composables/session';
+import { useSettings } from '@/composables/settings';
 import { isPeriodAllowed } from '@/store/settings/utils';
+import { useStore } from '@/store/utils';
 import { LAST_KNOWN_TIMEFRAME } from '@/types/frontend-settings';
 
 const validator = (value: any) =>
   Object.values(TimeFramePeriod).includes(value) ||
   value === TimeFramePersist.REMEMBER;
 
-@Component({
-  components: { TimeframeSelector, Fragment },
-  methods: {
-    ...mapMutations('session', ['setTimeframe'])
-  }
-})
-export default class TimeFrameSettings extends Mixins(
-  PremiumMixin,
-  SettingsMixin
-) {
-  @Prop({ required: true, type: Object })
-  message!: { error: string; success: string };
-
-  @Prop({
-    required: true,
-    type: String,
-    validator: validator
-  })
-  value!: TimeFrameSetting;
-
-  @Prop({
-    required: true,
-    type: Array
-  })
-  visibleTimeframes!: TimeFramePeriod[];
-
-  @Prop({
-    required: true,
-    type: String,
-    validator: validator
-  })
-  currentSessionTimeframe!: TimeFramePeriod;
-  setTimeframe!: (timeframe: TimeFramePeriod) => void;
-
-  get appendedVisibleTimeframes() {
-    return [TimeFramePersist.REMEMBER, ...this.visibleTimeframes] as const;
-  }
+export default defineComponent({
```

Comment:
Out of curiosity is there any reason why for some of the components the setup is used and for some the `defineComponent`?

Reply:
I want to use setup for all of them, but unfortunately the `defineProps` and `defineEmits` don't works as expected. So components that have props and emits, I use `defineComponent`